src/ForeignKeyDelegate.cpp

```diff
@@ -0,0 +1,156 @@
+#include "sqlitedb.h"
+#include "ForeignKeyDelegate.h"
+
+#include <QRegularExpressionMatch>
+#include <QComboBox>
+#include <QLineEdit>
+#include <QPushButton>
+#include <QHBoxLayout>
+
+class ForeignKeyEditor : public QWidget
+{
+    Q_OBJECT
+
+public:
+    ForeignKeyEditor(QWidget* parent = Q_NULLPTR)
+        : QWidget(parent)
+        , tablesComboBox(new QComboBox(this))
+        , idsComboBox(new QComboBox(this))
+        , m_btnReset(new QPushButton("&Reset", this))
+        , m_clauseEdit(new QLineEdit(this))
+    {
+        m_clauseEdit->setPlaceholderText("foreign key clauses(ON UPDATE, ON DELETE etc.)");
+
+        QHBoxLayout* layout = new QHBoxLayout(this);
+        layout->addWidget(tablesComboBox);
+        layout->addWidget(idsComboBox);
+        layout->addWidget(m_clauseEdit);
+        layout->addWidget(m_btnReset);
+        setLayout(layout);
+
+        connect(m_btnReset, &QPushButton::clicked, [&]
+        {
+            tablesComboBox->setCurrentIndex(-1);
+            idsComboBox->setCurrentIndex(-1);
+            m_clauseEdit->clear();
+        });
+
+        connect(tablesComboBox, static_cast<void(QComboBox::*)(int)>(&QComboBox::currentIndexChanged),
+                [=](int)
+        {
+            idsComboBox->setCurrentIndex(-1);
+        });
+
+        connect(idsComboBox, static_cast<void(QComboBox::*)(int)>(&QComboBox::currentIndexChanged),
+                [=](int index)
+        {
+            bool enableClausesEditor = (index!= -1);
+            m_clauseEdit->setEnabled(enableClausesEditor);
+        });
+    }
+
+    QString getResult() const
+    {
+        if (tablesComboBox->currentText().isEmpty() || idsComboBox->currentText().isEmpty())
+            return QString();
+        else
+            return QString("`%1`(`%2`) %3")
```

Comment:
Better to remove the ``` ` ``` quotes here and call sqlb::escapeIdentifier() instead. This handles quoting of the quote char correctly and makes it easier to change the quote char in the future if need arises.